### PR TITLE
HPCC-24313 Fix xmllib and esdllib build issues for ClientTools

### PR DESCRIPTION
**Pull request by mayx**
- Remove unnecessary dependency
- Comment out unsupported implementation for windows

Signed-off-by: Yanrui Ma <yanrui.ma@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by mayx** [comment]
@xwang2713 please review

**Comment by xwang2713** [comment]
Code looks good. Let me try target link remove does not impact Linux Platform and Clienttools builds

**Comment by xwang2713** [comment]
Linux Platform and Clienttools run OK. But more error reported from Windows Clienttools build:
cmake options: -G "Visual Studio 14 2015" -D CMAKE_BUILD_TYPE=Release -DINCLUDE_EE_PLUGINS=ON -DUSE_AWS=OFF -DUSE_NATIVE_LIBRARIES=OFF -DCHECK_GIT_TAG=0 -DCLIENTTOOLS_ONLY=ON -DSIGN_DIRECTORY=c:/hpcc/sign -DEXTERNALS_DIRECTORY=c:/hpcc/platform_externals_vs2015 -DUSE_APR=OFF -DUSE_CASSANDRA=OFF -DUSE_CBLAS=OFF -DWITH_PLUGINS=OFF -DUSE_AZURE=OFF

errors: (more than 200 errors. just list top ones):
22:03:04      Creating library D:/J/workspace/CTEEW32-J24313/CE/W19-301/build/bin/Release/xmllib.lib and object D:/J/workspace/CTEEW32-J24313/CE/W19-301/build/bin/Release/xmllib.exp
22:03:04 libxslt_processor.obj : error LNK2019: unresolved external symbol _xmlInitMemory referenced in function "public: __thiscall CLibXslProcessor::CLibXslProcessor(void)" (??0CLibXslProcessor@@QAE@XZ) [D:\J\workspace\CTEEW32-J24313\CE\W19-301\build\system\xmllib\xmllib.vcxproj]
22:03:04 libxslt_processor.obj : error LNK2019: unresolved external symbol _xmlCleanupMemory referenced in function "public: virtual __thiscall CLibXslProcessor::~CLibXslProcessor(void)" (??1CLibXslProcessor@@UAE@XZ) [D:\J\workspace\CTEEW32-J24313\CE\W19-301\build\system\xmllib\xmllib.vcxproj]
22:03:04 libxslt_processor.obj : error LNK2019: unresolved external symbol _xmlFreeDoc referenced in function "public: virtual __thiscall CLibXmlSource::~CLibXmlSource(void)" (??1CLibXmlSource@@UAE@XZ) [D:\J\workspace\CTEEW32-J24313\CE\W19-301\build\system\xmllib\xmllib.vcxproj]
22:03:04 libxml_xpathprocessor.obj : error LNK2001: unresolved external symbol _xmlFreeDoc [D:\J\workspace\CTEEW32-J24313\CE\W19-301\build\system\xmllib\xmllib.vcxproj]
22:03:04 libxslt_processor.obj : error LNK2019: unresolved external symbol _xmlInitParser referenced in function "public: __thiscall CLibXslProcessor::CLibXslProcessor(void)" (??0CLibXslProcessor@@QAE@XZ) [D:\J\workspace\CTEEW32-J24313\CE\W19-301\build\system\xmllib\xmllib.vcxproj]
22:03:04 libxslt_processor.obj : error LNK2019: unresolved external symbol _xmlCleanupParser referenced in function "public: virtual __thiscall CLibXslProcessor::~CLibXslProcessor(void)" (??1CLibXslProcessor@@UAE@XZ) [D:\J\workspace\CTEEW32-J24313\CE\W19-301\build\system\xmllib\xmllib.vcxproj]
22:03:04 libxslt_processor.obj : error LNK2019: unresolved external symbol _xmlParseFile referenced in function "public: struct _xmlDoc * __thiscall CLibXmlSource::getParsedXml(void)" (?getParsedXml@CLibXmlSource@@QAEPAU_xmlDoc@@XZ) [D:\J\workspace\CTEEW32-J24313\CE\W19-

**Comment by xwang2713** [comment]
Tested latest change.  The build run successfully.

File: esp/esdllib/CMakeLists.txt [hunk 1]
```diff
@@ -73,7 +73,4 @@ IF (USE_LIBXSLT)
     include_directories (
          ${LIBXML2_INCLUDE_DIR}
     )
-    target_link_libraries ( esdl_svc_engine
```
Comment (richardkchapman) [inline esp/esdllib/CMakeLists.txt:76]: Are the includes still needed, or should the whole if block be removed?

Comment (mayx) [reply esp/esdllib/CMakeLists.txt:76]: The includes are still needed, I'll look at whether it's possible to remove it.

**Pull request merged**